roaster-jaxrs/src/main/java/hu/icellmobilsoft/roaster/jaxrs/response/ResponseProcessor.java

```diff
@@ -0,0 +1,231 @@
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.Invocation.Builder;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import javax.ws.rs.core.UriBuilder;
+
+import org.eclipse.microprofile.config.ConfigProvider;
+import org.jboss.resteasy.client.jaxrs.ResteasyClient;
+import org.jboss.resteasy.client.jaxrs.ResteasyWebTarget;
+import org.jboss.resteasy.plugins.providers.multipart.MultipartFormDataOutput;
+import org.jboss.resteasy.plugins.providers.multipart.MultipartFormDataWriter;
+
+import hu.icellmobilsoft.coffee.module.mp.restclient.provider.DefaultBaseExceptionResponseExceptionMapper;
+import hu.icellmobilsoft.coffee.module.mp.restclient.provider.DefaultLoggerClientRequestFilter;
+import hu.icellmobilsoft.coffee.module.mp.restclient.provider.DefaultLoggerClientResponseFilter;
+import hu.icellmobilsoft.roaster.api.TestException;
+
+/**
+ * Base Response REST handler based on JXRS
+ *
+ * @param <RESPONSE>
+ *            response class (any type)
+ * @author imre.scheffer
+ * @since 0.8.0
+ */
+public abstract class ResponseProcessor<RESPONSE> {
+
+    /**
+     * Base URI config key
+     *
+     * @return config key like "project.service.base.uri"
+     */
+    public abstract String baseUriKey();
+
+    /**
+     * Get value by {@link #baseUriKey()} from microprofile config
+     *
+     * @return value like "http://localhost:8080"
+     */
+    public String baseUri() {
+        return ConfigProvider.getConfig().getValue(baseUriKey(), String.class);
+    }
+
+    /**
+     * HTTP path to call
+     *
+     * @return value like "/test/service/generate/testData"
+     */
+    public abstract String path();
+
+    /**
+     * Expected REST response status code<br>
+     * <br>
+     * Default value is <strong>{@value Status#OK}</strong>
+     */
+    private int expectedStatusCode = Status.OK.getStatusCode();
+
+    /**
+     * Call and get JSON object from HTTP GET method
+     *
+     * @param responseClass
+     *            response class
+     * @param pathParams
+     *            The path parameters.
+     * @return response object cast to responseClass
+     */
+    public RESPONSE getOctetStream(Class<RESPONSE> responseClass, Object... pathParams) {
+        // REST kliens megszerzese
+        ResteasyClient client = (ResteasyClient) ClientBuilder.newClient();
+        ResteasyWebTarget target = client.target(UriBuilder.fromUri(baseUri() + path()).build(pathParams));
+        target.register(MultipartFormDataWriter.class);
+        CDI<Object> cdi = CDI.current();
+        // ez csak MP rest clientnel tud mukodni
+        // target.register(RoasterRestClientBuilderListener.class);
+        target.register(cdi.select(DefaultLoggerClientRequestFilter.class).get());
+        target.register(cdi.select(DefaultLoggerClientResponseFilter.class).get());
+        // GET kuldese
+        Builder builder = target.request().accept(MediaType.APPLICATION_JSON_TYPE);
```

Comment:
Itt nem octet-stream kell?

Reply:
De koszi, a tesztek ezt nem hoztak ki mivel kamu a server...

Reply:
Az excpectation-be bele lehet rakni.